src/components/plots/gridPlot.ts

```diff
@@ -55,15 +60,32 @@ export module Plot {
     }
 
     protected _generateAttrToProjector() {
+
       var attrToProjector = super._generateAttrToProjector();
-      var xStep = (<Scale.Ordinal> this._xScale).rangeBand();
-      var yStep = (<Scale.Ordinal> this._yScale).rangeBand();
-      attrToProjector["width"]  = () => xStep;
-      attrToProjector["height"] = () => yStep;
-      var xAttr = attrToProjector["x"];
-      var yAttr = attrToProjector["y"];
-      attrToProjector["x"] = (d, i, u, m) => xAttr(d, i, u, m) - xStep / 2;
-      attrToProjector["y"] = (d, i, u, m) => yAttr(d, i, u, m) - yStep / 2;
+
+      // Copy each of the different projectors
+      var x1Attr = attrToProjector["x"];
+      var y1Attr = attrToProjector["y"];
+      var x2Attr = attrToProjector["x2"];
+      var y2Attr = attrToProjector["y2"];
+
+      // Adjust the xScale if it is ordinal
+      if (this._xScale instanceof Scale.Ordinal) {
+        attrToProjector["width"] = () => (<Scale.Ordinal> this._xScale).rangeBand();
+        attrToProjector["x"] = (d, i, u, m) => x1Attr(d, i, u, m) - attrToProjector["width"](d, i, u, m) / 2;
+      } else {
+        attrToProjector["width"] = (d, i, u, m) => Math.abs(x2Attr(d, i, u, m) - x1Attr(d, i, u, m));
+      }
+
+      // Adjust the yScale if it is ordinal
+      if (this._yScale instanceof Scale.Ordinal) {
+        attrToProjector["height"] = () => (<Scale.Ordinal> this._yScale).rangeBand();
+        attrToProjector["y"] = (d, i, u, m) => y1Attr(d, i, u, m) - attrToProjector["height"](d, i, u, m) / 2;
+      } else {
+        attrToProjector["height"] = (d, i, u, m) => Math.abs(y2Attr(d, i, u, m) - y1Attr(d, i, u, m));
+        attrToProjector["y"] = (d, i, u, m) => y1Attr(d, i, u, m) - attrToProjector["height"](d, i, u, m);
+      }
```

Comment:
What we might want to do for this project class is correct x1/y1/x2/y2 and then hand these off to the Drawer to compute what width/height/x/y should be